vowpalwabbit/parser.cc

```diff
@@ -1005,47 +960,35 @@ size_t get_feature_number(example* ec) { return ec->num_features; }
 float get_confidence(example* ec) { return ec->confidence; }
 }  // namespace VW
 
-void initialize_examples(vw& all)
+example* example_initializer::operator()(example* ex)
 {
-  all.p->used_index = 0;
-  all.p->begin_parsed_examples = 0;
-  all.p->end_parsed_examples = 0;
-  all.p->done = false;
-
-  all.p->examples = calloc_or_throw<example>(all.p->ring_size);
-
-  for (size_t i = 0; i < all.p->ring_size; i++)
-  {
-    memset(&all.p->examples[i].l, 0, sizeof(polylabel));
-    all.p->examples[i].in_use = false;
-  }
+  memset(&ex->l, 0, sizeof(polylabel));
+  ex->in_use = false;
+  ex->passthrough = nullptr;
+  ex->tag = v_init<char>();
+  ex->indices = v_init<namespace_index>();
+  memset(&ex->feature_space, 0, sizeof(ex->feature_space));
+  return ex;
 }
 
-void adjust_used_index(vw& all) { all.p->used_index = all.p->begin_parsed_examples; }
+void adjust_used_index(vw& all) { /* no longer used */ }
```

Comment:
Can we just remove this function if it's a noop?

Reply:
Yes I'd like to, I just need to get around to it since it is used in quite a few bindings